src/User/Command/EditUserHandler.php

```diff
@@ -109,15 +110,21 @@ public function handle(EditUser $command)
         if (isset($relationships['groups']['data']) && is_array($relationships['groups']['data'])) {
             $actor->assertPermission($actor->can('editGroups', $user));
 
+            $oldGroups = $user->groups()->get()->all();
+            $oldGroupIds = Arr::pluck($oldGroups, 'id');
+
             $newGroupIds = [];
             foreach ($relationships['groups']['data'] as $group) {
                 if ($id = Arr::get($group, 'id')) {
                     $newGroupIds[] = $id;
                 }
             }
 
+            $adminChanged = in_array('1', array_diff($oldGroupIds, $newGroupIds)) || in_array('1', array_diff($newGroupIds, $oldGroupIds));
```

Comment:
Should this be moved out to a guard (like SelfDemotionGuard)? We decided to leave it here since that felt simpler. Alternatively, we could move SelfDemotionGuard here for simplicity?

Reply:
I don't feel strongly about this, but I wouldn't remove the SelfDemotionGuard either, maybe just leave things as is ?